openvpn-wire-protocol.xml

```diff
@@ -1542,6 +1544,226 @@ struct datakeys {
         PRF.
       </t>
     </section>
+    <section anchor="epochkeys" title="OpenVPN epoch data channel keys">
+      <section title="AEAD cipher usage limits">
+      <t>
+      Recent advances in cryptography research have shown that GCM ciphers, most notably AES-GCM have
+      theoretical vulnerabilities when AES-GCM is used with the same key with too many block encryption
+      invocations. See <xref target="usage-limits-aead-draft">Usage Limits on AEAD Algorithms</xref>  and the 
+      referenced research papers in that document like <xref target="AEAD-LIMITS"/> for more information.
+    </t>
+    <t>
+      To avoid using the same encryption key for too many packets without requiring to do a full renegotiation
+      OpenVPN uses epoch data channel keys. The term epoch is used to avoid confusion with the already
+      existing term key-id and since <xref target="RFC9147">DTLS 1.3</xref> uses the same terminology.
+    </t>
+    <t>
+      A sender should calculate a safe amount of packets to send/number of encryptions that can be done.
+      The actual number depends on the cipher being used.  From <xref target="usage-limits-aead-draft"> current
+      standards</xref>, there is no limit to consider for Chacha20-Poly1305, so a sender can use the full
+      48 bit of the epoch counter for encrypting packets before moving to a new epoch and encryption key.
+      For AES-GCM, the limit is 2^32.5 when using a packet size of 1500 using this document and the safety
+      margin of 2^-50 but the   <xref target="RFC8446">TLS 1.3</xref> standard uses a higher safety margin of 2^-57,
+      which translates to 2^24.5 for maximum TLS record sizes (2^14 or 16kB).  OpenVPN implementations should
+      choose the same (higher) safety margin as TLS.
+    </t>
+    <t>
+      In order to guarantee that safety marging an implementation has to ensure that the number of plain text
+      blocks s and the number of packets q satisfies  the following equation:
+
+      <artwork>
+        q + s &le; p^(1/2) * 2^(129/2) - 1
+      </artwork>
+      Using our confidentially margin of 2^-57, this gives us:
+
+      <artwork>
+        q + s &le; 2^36
+      </artwork>
+
+      Tracking the number of blocks of encrypted clear text data limit this allows not having to fix L or
```

Comment:
I don't get this sentence. `Tracking the number of blocks of clear text data` and next `limit this allows not having to `. Can you rephrase it?

Reply:
Yes, I rephrased the whole section to make it easier to understand